observationtree/src/main/scala/explore/observationtree/ConstraintSetObsQueries.scala

```diff
@@ -0,0 +1,204 @@
+import monocle.Getter
+import monocle.macros.Lenses
+
+object ConstraintSetObsQueries {
+
+  @Lenses
+  case class ConstraintSetIdName(id: ConstraintSet.Id, name: NonEmptyString)
+
+  object ConstraintSetIdName {
+    implicit val decoder: Decoder[ConstraintSetIdName] = new Decoder[ConstraintSetIdName] {
+      final def apply(c: HCursor): Decoder.Result[ConstraintSetIdName] =
+        for {
+          id   <- c.downField("id").as[ConstraintSet.Id]
+          name <- c.downField("name")
+                    .as[Option[String]]
+                    .map(
+                      _.flatMap(name => NonEmptyString.from(name).toOption)
+                        .getOrElse(Constants.UnnamedConstraintSet)
+                    )
+        } yield ConstraintSetIdName(id, name)
+    }
+  }
+
+  // @Lenses
+  // case class ObsForConstraintSet(
+  //   id:              Observation.Id,
+  //   name:            Option[String],
+  //   constraintSetId: ConstraintSet.Id
+  // )
+
+  type ConstraintSetList = KeyedIndexedList[ConstraintSet.Id, ConstraintSetIdName]
+  type ObsList           = KeyedIndexedList[Observation.Id, ObsSummary]
+
+  @Lenses
+  case class ConstraintSetsWithObs(constraintSets: ConstraintSetList, obs: ObsList)
+
+  @GraphQL(debug = false)
+  object ConstraintSetsObsQuery extends GraphQLOperation[ObservationDB] {
+    val document = s"""
+      query {
+        constraintSets(programId: "p-2", first: ${Int.MaxValue}) {
+          nodes {
+            id
+            name
+          }
+        }
+
+        observations(programId: "p-2", first: ${Int.MaxValue}) {
+          nodes {
+            id
+            name
+            observationTarget {
+              type: __typename
+              ... on Target {
+                target_id: id
+              }
+              ... on Asterism {
+                asterism_id: id
+              }
+            }
+            constraintSet {
+              id
+            }            
+          }
+        }    
+      }
+    """
+
+    object Data {
+      object ConstraintSets {
+        type Nodes = ConstraintSetIdName
+      }
+
+      object Observations {
+        type Nodes = ObsSummary
+      }
+
+      val asConstraintSetsWithObs: Getter[Data, ConstraintSetsWithObs] = data => {
+        ConstraintSetsWithObs(
+          KeyedIndexedList.fromList(data.constraintSets.nodes, ConstraintSetIdName.id.get),
+          KeyedIndexedList.fromList(data.observations.nodes, ObsSummary.id.get)
+        )
+      }
+    }
+  }
+
+  @GraphQL
+  object CostraintSetEditSubscription extends GraphQLOperation[ObservationDB] {
+    val document = """
+      subscription {
+        constraintSetEdit(programId: "p-2") {
+          id
+        }
+      }
+    """
+  }
+
+  @GraphQL
+  object ObservationEditSubscription extends GraphQLOperation[ObservationDB] {
+    val document = """
+      subscription {
+        observationEdit(programId: "p-2") {
+          id
+        }
+      }    
+    """
+  }
```

Comment:
This is the same as `ObsQueries.ProgramObservationsEditSubscription`. We should reuse that one. It's also duplicate in `TargetObsQueries`, let's remove it from there too.